packages/plugin-core/src/components/views/PreviewLinkHandler.ts

```diff
@@ -50,103 +58,96 @@ export class PreviewLinkHandler implements IPreviewLinkHandler {
             column: vscode.ViewColumn.One,
-            anchor: noteData.anchor,
-          });
-
-          return;
-        } catch (err) {
-          Logger.error({ error: ErrorFactory.wrapIfNeeded(err) });
-          return;
-        }
-      }
-      case LinkType.WEBSITE: {
-        // Updated preview appears to already open the external links in the browser by itself
-        // Hence running `VSCodeUtils.openLink(data.href!);` causes double opening
-        // of the link within the browser.
-        return;
-      }
-      case LinkType.MARKDOWN: {
-        // assume local note - open relative to current vault
-        const note = this._ext.getEngine().notes[data.id!];
-        if (!note) return;
-
-        // noteId in this case is name of file
-        const fullPath = path.join(
-          this._ext.getEngine().wsRoot,
-          VaultUtils.getRelPath(note.vault),
-          note.id
+          }
         );
-        return VSCodeUtils.openLink(fullPath);
-      }
-      default: {
-        // default, do nothing. let vscode link handler handle it
-        return;
+        if (!_.isEmpty(uri.fragment) && editor) {
+          const anchor = AnchorUtils.string2anchor(uri.fragment);
+          await GotoNoteCommand.trySelectRevealNonNoteAnchor(editor, anchor);
+        }
+        return LinkType.TEXT;
+      } else {
+        // Otherwise it's a binary file, try to open it with the default program
+        ShowPreviewAssetOpener.openWithDefaultApp(path.normalize(fullPath));
+        return LinkType.ASSET;
       }
     }
+    // If nothing applies, VSCode's default will hopefully handle it
+    Logger.debug({
+      ctx,
+      msg: "Nothing applied for the URL, had to fall back to VSCode default.",
+    });
+    return LinkType.UNKNOWN;
   }
 
-  public classifyLink({ href }: NoteViewMessage["data"]): LinkType {
-    if (
-      href &&
-      href.startsWith("vscode-webview") &&
-      href.includes("/assets/")
-    ) {
-      // Note: currently even when the wiki link is fully vault qualified as example
-      // of [[dendron://assets/note-in-asset-vault]] When it is clicked within the preview
-      // the href will look along the lines of:
-      // `vscode-webview://72db5b4c-61f8-400b-808c-771184cb3d7f/r68Zw7OChUZWvbD10qqmY`
-      // href will contain the id of the note but it will NOT contain the vault
-      // hence we should avoid the issue of parsing 'assets' vault name even if someone names their
-      // vault 'assets'.
-      return LinkType.ASSET;
-    } else if (href && href.startsWith("vscode-webview")) {
-      return LinkType.WIKI;
-    } else if (
-      href &&
-      (href.startsWith("http://") || href.startsWith("https://"))
-    ) {
-      return LinkType.WEBSITE;
-    } else {
-      return LinkType.UNKNOWN;
-    }
-  }
-
+  /** Returns a note if one was found, undefined if no notes were found, and null if the link was ambiguous and user cancelled the prompt to pick a note. */
```

Comment:
nit: comment runs long. 

As an aside, I like this vs code ext which can help auto-flow comments to fit inside a specified length: https://marketplace.visualstudio.com/items?itemName=stkb.rewrap

Reply:
Thanks for the suggestion, I do have it installed actually! It was something I missed a lot moving from Emacs to VSCode.